source/types.ts

```diff
@@ -25,43 +25,39 @@ export interface CreateComponentOptions {
   onAborted?: () => unknown;
 }
 
-export interface Entity {
-  id: string;
-  __entity_type__: string;
-}
 interface ResponseMetadata {
   next: {
     offset: number | null;
   };
 }
 export interface SearchOptions {
   expression: string;
-  entityType: string;
+  entityType: EntityType;
   terms?: string[];
   contextId?: string;
   objectTypeIds?: string[];
 }
 
-export interface QueryResponse<T = Data> {
-  data: T[];
+export interface QueryResponse<K extends EntityType = EntityType> {
+  data: EntityData<K>[]; // Later this needs to take multiple different K and give the correct types
   action: "query";
   metadata: ResponseMetadata;
 }
 
-export interface CreateResponse<T = Data> {
-  data: T;
+export interface CreateResponse<K extends EntityType = EntityType> {
```

Comment:
For consistency, change this to `TEntityType` as well

Reply:
I prefer the distinction between string keys and types, I feel it makes it easier to understand what kind of generic is expected. But we should be consistent, how about changing them all to KEntityType?

Reply:
didn't know that was a convention. it's new to me so i don't like it ;)

TEntityType > KEntityType > K

both of TEntityType and KEntityType are acceptable!

Reply:
@ffMathy - want to be the deciding vote between TEntityType and KEntityType?

Reply:
I have no preference, sorry 🤣

One sec, let me throw a coin...

Okay, it's TEntityType 🙏👌